src/Utilities/Compiler/Options/SymbolNamesOption.cs

```diff
@@ -12,23 +15,32 @@ namespace Analyzer.Utilities
 {
     internal sealed class SymbolNamesOption : IEquatable<SymbolNamesOption?>
     {
+        private const SymbolKind AllKinds = SymbolKind.ErrorType;
+        private const char WildcardChar = '*';
+
+        private static readonly KeyValuePair<string, string?> NoWildcardMatch = default;
+
         public static readonly SymbolNamesOption Empty = new SymbolNamesOption();
 
         private readonly ImmutableDictionary<string, string?> _names;
         private readonly ImmutableDictionary<ISymbol, string?> _symbols;
+        private readonly ImmutableDictionary<SymbolKind, ImmutableDictionary<string, string?>> _wildcardNamesBySymbolKind;
```

Comment:
I can't have `null` as the generic entry to try to match the various symbol kinds so I have added a constant to represent this state using the `ErrorType`.

Reply:
Can you add a doc comment with examples for the different combinations of key-value pairs that can show up in this dictionary?